components/org.wso2.carbon.securevault/src/main/java/org/wso2/carbon/securevault/SecureVaultInitializer.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.securevault;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.securevault.exception.SecureVaultException;
+import org.wso2.carbon.securevault.internal.SecureVaultConfigurationProvider;
+import org.wso2.carbon.securevault.internal.SecureVaultDataHolder;
+import org.wso2.carbon.securevault.internal.SecureVaultImpl;
+import org.wso2.carbon.securevault.model.SecureVaultConfiguration;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Optional;
+
+/**
+ * Responsible for initializing secure vault.
+ *
+ * @since 5.2.0
```

Comment:
version mentioned is wrong. This is the kernel version. This should be changed to the carbon-secvault version it being introduced.